test/unit/crash-reporter.js

```diff
@@ -0,0 +1,181 @@
+    done();
+  },
+
+  tearDown: function(done) {
+    this.sandbox.restore();
+    done();
+  },
+
+  onSuccess: function(test) {
+    test.expect(1);
+
+    CrashReporter.on().then(() => {
+      test.equal(this.pWrite.callCount, 1);
+      test.done();
+    });
+  },
+
+  onFailure: function(test) {
+    test.expect(2);
+    this.pWrite.restore();
+    this.pWrite = this.sandbox.stub(Preferences, 'write').returns(Promise.reject());
+
+    // Despite the write failure, we don't _want_ to crash the crash reporter
+    CrashReporter.on().then(() => {
+      test.equal(this.pWrite.callCount, 1);
+      test.equal(this.logsErr.callCount, 1);
+      test.done();
+    });
+  },
+
+};
+
+exports['CrashReporter.off'] = {
+  setUp: function(done) {
+    this.sandbox = sinon.sandbox.create();
+    this.logsErr = this.sandbox.stub(logs, 'err');
+    this.pWrite = this.sandbox.stub(Preferences, 'write').returns(Promise.resolve());
+    done();
+  },
+
+  tearDown: function(done) {
+    this.sandbox.restore();
+    done();
+  },
+
+  offSuccess: function(test) {
+    test.expect(1);
+
+    CrashReporter.off().then(() => {
+      test.equal(this.pWrite.callCount, 1);
+      test.done();
+    });
+  },
+
+  offFailure: function(test) {
+    test.expect(2);
+    this.pWrite.restore();
+    this.pWrite = this.sandbox.stub(Preferences, 'write').returns(Promise.reject());
+
+    // Despite the write failure, we don't _want_ to crash the crash reporter
+    CrashReporter.off().then(() => {
+      test.equal(this.pWrite.callCount, 1);
+      test.equal(this.logsErr.callCount, 1);
+      test.done();
+    });
+  },
+
+};
+
+exports['CrashReporter.submit'] = {
+  setUp: function(done) {
+    this.sandbox = sinon.sandbox.create();
+    this.logsErr = this.sandbox.stub(logs, 'err');
+    this.logsInfo = this.sandbox.stub(logs, 'info');
+    this.pRead = this.sandbox.stub(Preferences, 'read').returns(Promise.resolve('on'));
+    this.pWrite = this.sandbox.stub(Preferences, 'write').returns(Promise.resolve());
+    this.crPost = this.sandbox.spy(CrashReporter, 'post');
+    this.request = this.sandbox.stub(request, 'post', (opts, handler) => {
+      return handler(null, {}, '{}');
+    });
+    done();
+  },
+
+  tearDown: function(done) {
+    this.sandbox.restore();
+    done();
+  },
+
+  submit: function(test) {
+    test.expect(7);
+
+    var report = 'Error: undefined is not a function';
+    CrashReporter.submit(report).then(() => {
+      var args = this.crPost.lastCall.args;
+      test.equal(this.crPost.callCount, 1);
+      test.equal(typeof args[0], 'string');
+      test.ok(args[0].indexOf('CLI version') > 0, 'Label CLI version should be present');
+      test.ok(args[0].indexOf('Node version') > 0, 'Label Node version should be present');
+      test.ok(args[0].indexOf('OS platform') > 0, 'Label OS platform should be present');
+      test.ok(args[0].indexOf('OS release') > 0, 'Label OS release should be present');
```

Comment:
`args[0]` is actually `labels` and 
```javascript
typeof args[0] === 'string'
```
So, I cant use `args[0].includes(...)`.

---

Reference is in the source for [submit crash handler](https://github.com/tessel/t2-crash-reporter/blob/master/main.py#L73)